### PR TITLE
Holiday Prefills

### PR DESCRIPTION
**Pull request by batemapf**
## Description

Allows for the pre-filling of timecards with "holidays" (or any other project(s)). Accomplishes this by:
- Creating a new model class, HolidayPrefills;
- Linking HolidayPrefills to reporting periods;
- Adding logic to the timecard form initialization process that checks to see if the reporting period linked to the timecard has a holiday, and if so, pre-fills the timecard with the project and number of hours.

Addresses #426 and #454. 

## Additional information

Adding a pre-fill item to a reporting period:

screen shot 2016-11-25 at 4 30 57 pm

When the user opens their timecard for the first time for that reporting period:

screen shot 2016-11-25 at 4 32 11 pm

### CODE REVIEW

**Comment by batemapf** [comment]
OK @annalee this one is ready for your 👀

File: tock/hours/views.py [hunk 1]
```diff
@@ -673,17 +670,39 @@ def last_timecard(self):
             '-reporting_period__start_date',
         ).first()
 
-    def prefilled_formset(self, timecard):
-        project_ids = set(
-            tco.project_id for tco in
-            timecard.timecardobjects.all()
-        )
-        extra = len(project_ids) + 1
+    def prefilled_formset(self):
```
Comment (brendansudol) [inline tock/hours/views.py]: a little bit cleaner / tighter, take or leave:

```
def prefilled_formset(self):
    project_ids, extra = [], 1
    if self.last_timecard():
        project_ids = set(
            tco.project_id for tco in
            timecard.timecardobjects.all()
        )
        extra = len(project_ids) + 1
    
    rp = ReportingPeriod.objects \
        .prefetch_related('holiday_prefills__project') \
        .get(start_date=self.kwargs['reporting_period'])

    init = []
    if rp.holiday_prefills:
        init += [
            {'hours_spent': hp.hours_per_period, 'project': hp.project.id}
            for hp in rp.holiday_prefills.all()
        ]
    init += [{'hours_spent': None, 'project': pid} for pid in project_ids]

    formset = timecard_formset_factory(extra=extra)
    return formset(initial=init)
```

**Comment by brendansudol** [comment]
overall looks good, have a good small q's, but let's go through in person tomorrow

**Comment by brendansudol** [comment]
looks great @batemapf, and nice useful new feature 🚢 💥

**Pull request merged**